### ISSUE TITLE
libostree assertion hit when trying to install a non-ostree image on a OSTree primary

### ISSUE THREAD

**Issue by martingkelly** [advancedtelematic/aktualizr#355]
Hi,

I tried the quickstart guide here: https://docs.atsgarage.com/quickstarts/qemuvirtualbox.html
using the repo here: https://github.com/advancedtelematic/updater-repo.git

After attempting to create a deployment through ATS garage (using a small test file), I'm finding that the deployment doesn't complete. I found this line in /var/log/messages:

`
Sep 20 20:37:44 qemux86-64 daemon.info aktualizr[814]: OSTree:ERROR:src/libostree/ostree-core.c:1237:ostree_checksum_inplace_to_bytes: assertion failed: (big != -1)
`

**Comment by cajun-rat** [comment]
That's a new one on me.  Before I dig too deeply into the code, could you just check that it isn't running out of disk space?

**Comment by martingkelly** [comment]
Yep, disk space looks alright:

```
root@qemux86-64:~# df -h
Filesystem                Size      Used Available Use% Mounted on
df: /run/initramfs: No such file or directory
devtmpfs                494.3M         0    494.3M   0% /dev
df: /run/initramfs/sys: No such file or directory
tmpfs                   496.2M         0    496.2M   0% /dev/shm
df: /run/initramfs/tmp: No such file or directory
df: /run/initramfs/run: No such file or directory
/dev/hda                238.9M    130.0M     92.4M  58% /sysroot
/dev/hda                238.9M    130.0M     92.4M  58% /
/dev/hda                238.9M    130.0M     92.4M  58% /var
/dev/hda                238.9M    130.0M     92.4M  58% /boot
/dev/hda                238.9M    130.0M     92.4M  58% /usr
tmpfs                   496.2M      8.3M    487.8M   2% /run
tmpfs                   496.2M         0    496.2M   0% /sys/fs/cgroup
tmpfs                   496.2M         0    496.2M   0% /tmp
tmpfs                   496.2M         0    496.2M   0% /var/volatile
```

**Comment by cajun-rat** [comment]
Hmm, OK. We are trying to reproduce it here: Is there anything 'unusual' about the file you added?

**Comment by martingkelly** [comment]
Nothing that I can think of. I tried both a 22-byte text file and an image built by the update-repo yocto recipes. I can send you the files you tried, or the image I'm using or similar. I think are in my Garage account as well.

**Comment by cajun-rat** [comment]
Do you still have the duff image around?  If so, could you send me (phil@advancedtelematic.com) the contents of both targets.json files, please?

~~~
# find /var/sota/ -iname targets.json
/var/sota/metadata/repo/targets.json
/var/sota/metadata/director/targets.json
~~~

Looking at the ostree code, it appears to occur when converting a hex-formated hash into a list of bytes.  My guess is that perhaps it is the ostree commit hash getting corrupted somehow.

**Comment by martingkelly** [comment]
Sure, check your email :).

**Comment by cajun-rat** [comment]
We figured out that this is caused by trying to install a non-ostree image on a OSTree partition.

**Comment by cajun-rat** [comment]
This is fixed in #365. We now display a nice error message in this case.

The 'correct' solution for managing configuration files can be done in one of two ways: either by building an OSTree image with the right configuration for the device baked in (which works when the number of configurations is relatively low), or by treating configuration file as a simulated secondary ECU.

**Comment by martingkelly** [comment]
How would the "simulated secondary ECU" case work? Would that integrate well with tools?
